src/Exception/MalformedDsnException.php

```diff
@@ -0,0 +1,14 @@
+<?php
+
+namespace Doctrine\DBAL\Exception;
+
+use Doctrine\DBAL\Exception;
+
+/** @psalm-immutable */
+class MalformedDsnException extends Exception
```

Comment:
Have you seen https://github.com/doctrine/dbal/pull/5727? Should this exception extend LogicException? It's not meant to be checked, is it?

Reply:
Good catch, I'll change it.